presto-main/src/main/java/com/facebook/presto/sql/analyzer/UtilizedColumnsAnalyzer.java

```diff
@@ -526,7 +526,10 @@ public void copyFieldIdsToExploreForWithQuery(WithQuery withQuery)
             List<RelationId> relationIds = fieldsToExplore.keySet().stream()
                     .filter(key -> key.getSourceNode() instanceof Table && ((Table) key.getSourceNode()).getName().equals(name))
                     .collect(toImmutableList());
-            if (relationIds.size() != 1) {
+            if (relationIds.isEmpty()) {
```

Comment:
In what cases would the relationId would be empty?  Is that if the CTE doesn't actually get used in the query or are there other cases as well? Can you add tests?